cypress/integration/blocks.js

```diff
@@ -596,5 +596,35 @@ if (Cypress.env('API') !== 'guillotina') {
     //   //   cy.contains(expected);
     //   // }
     // });
+
+    it.only('Add Table of Contents block', () => {
+      const h2 = 'expectedH2';
+
+      //Add H2
+      cy.get('.block.inner.text .public-DraftEditor-content')
+        .type(` ${h2}`)
+        .setSelection(` ${h2}`);
+      cy.get(
+        '#page-edit .draftJsToolbar__buttonWrapper__1Dmqh:nth-of-type(5)',
+      ).click();
+      cy.get('.block.inner.text .public-DraftEditor-content')
+        .click()
+        .type(' {enter}');
+
+      //Add TOC Block
+      cy.get('.ui.basic.icon.button.block-add-button').click();
+      cy.get('.title')
+        .contains('Common')
+        .click();
+      cy.get('.ui.basic.icon.button.toc')
+        .contains('Table of Contents')
+        .click();
+
+      //Save
+      cy.get('#toolbar-save').click();
+
+      //View
+      cy.get(' a').contains(` ${h2}`);
```

Comment:
@steffenri this checks any "a" element. We should check the output of the TOC block for the h2 headline.

Reply:
Fixed that aswell. We now check for an a element in the list that gets created by the TOC Block. We also check for the exact header type.